app/controllers/api/news_controller.rb

```diff
@@ -88,4 +59,48 @@ def serialized(obj)
       formatted_data = format_item.call(data)
     end
   end
+
+  def process_news_feed(params)
```

Comment:
Hmmm, the name of this method makes me feel it would produce side effects (e.g. save actions). Maybe `apply_news_feed_filters(news_items, params)`